### ISSUE TITLE
can't initialize on Android and I'm stuck buffering.

### ISSUE THREAD

**Issue by Brightbong92** [doublesymmetry/react-native-track-player#2341]
**What is the need and use case of this feature?**

It sets up normally in ios, but not in android. It stays buffering. Why is that?

## ios log
<img width="730" alt="스크린샷 2024-07-09 오후 12 51 09" src="https://github.com/doublesymmetry/react-native-track-player/assets/51194127/807a9d04-14c9-4eaa-a114-2a4ab0a607f7">

## android log
<img width="723" alt="스크린샷 2024-07-09 오후 12 49 56" src="https://github.com/doublesymmetry/react-native-track-player/assets/51194127/00aadbaa-df2f-4966-b816-2a5eda4a3dbb">

## SetupService.ts
![스크린샷 2024-07-09 오후 12 50 21](https://github.com/doublesymmetry/react-native-track-player/assets/51194127/ef67a99b-b12c-4bb8-baa7-763e6cabd49a)

## App.ts
![스크린샷 2024-07-09 오후 12 50 41](https://github.com/doublesymmetry/react-native-track-player/assets/51194127/75cb4a48-f1ce-49a0-a038-27d8e751375d)

## usecase.ts
![스크린샷 2024-07-09 오후 12 52 25](https://github.com/doublesymmetry/react-native-track-player/assets/51194127/71bf220b-51e9-4eb0-966e-43bccedff0de)

![스크린샷 2024-07-09 오후 12 52 37](https://github.com/doublesymmetry/react-native-track-player/assets/51194127/2796f2ff-212d-458f-a24a-0afa6781021e)

**Comment by lovegaoshi** [comment]
try snipping the media src network request first with eg HTTPToolkit

**Comment by c-goettert** [comment]
I am facing the same issue (using react-native: 0.72.15). Downgrading to `4.0.0-rc04` fixes the issue for me which shows it's not a network issue...

**Comment by git-user-1337** [comment]
I have the same problem, does the current version work on Android for anyone?

**Comment by afiller** [comment]
We are having the same problem with RN 0.72.15. 😬 Any updates on this issue?

**Comment by c-goettert** [comment]
@Brightbong92 did you find any solution to this?

**Comment by Brightbong92** [comment]
@c-goettert 
I am sharing the method that I have solved.

```ts
// useSetupPalyer.ts
import { useEffect, useState } from 'react';

import TrackPlayer from 'react-native-track-player';

import { SetupService } from '@/utils/track-player';

export interface UseSetupPlayerProps {
  cb?: () => Promise<void>;
}

export function useSetupPlayer({ cb }: UseSetupPlayerProps) {
  const [isPlayerReady, setIsPlayerReady] = useState<boolean>(false);

  useEffect(() => {
    let unmounted = false;
    (async () => {
      await SetupService();
      if (unmounted) return;
      setIsPlayerReady(true);
      const queue = await TrackPlayer.getQueue();
      if (unmounted) return;
      if (queue.length <= 0) {
        cb && cb();
      }
    })();
    return () => {
      unmounted = true;
    };
  }, []);

  return isPlayerReady;
}

```

```ts
// useHooks.ts
const track1 = {
  url, // Load media from the network
  title: 'Avaritia',
  artist: 'deadmau5',
  album: 'while(1<2)',
  genre: 'Progressive House, Electro House',
  date: '2014-05-20T07:00:00+00:00', // RFC 3339
  artwork: 'https://example.com', // Load artwork from the network
  duration: 0, // Duration in seconds
};
...
  const addPlayer = useCallback(async () => {
    await TrackPlayer.add([track1]);
  }, []);

const isPlayerReady = useSetupPlayer({ cb: addPlayer });
...
return {
   ...
   isReady: isPlayerReady
   ...
};
```

```tsx
// view.tsx
...
const {isReady} = useHooks();
...
return (
  {
    isReady ? <Text>show</Text> : <Text>loading...</Text>
  }
)
```

**Comment by c-goettert** [comment]
Thanks for posting you solution @Brightbong92 . Indeed I found out that for me it was something else that has caused the bug: I was calling `TrackPlayer.reset()` immediately before starting the Player without waiting for it. Something like:
```
...
await TrackPlayer.setupPlayer()
...
TrackPlayer.reset()
...
await TrackPlayer.add([track])
TrackPlayer.play()
```

As the reset call is asynchronous it also removed the track I added to the queue immediately after the call. I could fix it by adding an `await` to the reset function (it returns a promise). This issue can be closed from my perspective.
